README.md

```diff
@@ -74,27 +65,55 @@ docker container run --rm --detach \
   pact-provider-states-setup:latest
 ```
 
-*Pact provider states setup* is now listening at `http://localhost:8000/`.
-
-### 4. Verify a Pact contract using provider state
-
-Run Pact provider verification using [Pact command line
-tools](https://docs.pact.io/implementation_guides/cli), like this:
+### 3. Run Your Pact Verification
+Tell your Pact verifier where to find us:
 
 ```sh
 pact verify \
   --provider-base-url='http://my-provider:12345/' \
   --provider-states-setup-url='http://localhost:8000/' \
-  /path/to/myconsumer-myprovider.json
+  /path/to/your-pact-contract.json
 ```
 
-Before each Pact contract interaction, the GraphQL mutation, passed as a Pact
-provider state, is first forwarded to your provider app before the Pact
-verification is performed.
+### 4. Using with Different GraphQL Operations
+You can use various types of GraphQL operations in your provider states. Here's another example using a query:
+
+```csharp
+.Given("Setup initial data", new Dictionary<string, string>
+{
+    {
+        "query",
+        @"
+        query {
+            SetupTestData(scenario: ""happy-path"") {
+                success
+                message
+            }
+        }
+        "
+    }
+})
+```
+
+## How It Works
+Think of us as your provider state concierge:
+1. You define your provider states in your consumer tests
+2. We catch the provider state setup requests
+3. We forward your GraphQL operations to your provider
+4. Your Pact verification runs smoothly with the right data in place
+
+## Typical Use Cases
+- Setting up test data before verification
+- Cleaning up data between tests
+- Configuring provider behavior for specific scenarios
+- Handling complex state requirements through GraphQL mutations
+
+## Contributing
+Got ideas? Found a bug? Want to add support for more operations? We love contributions! Check out our [Contributing Guide](CONTRIBUTING.md) to get started. Because making Pact testing better is a team sport! 🏆

```

Comment:
```suggestion
Got ideas? Found a bug? Want to add support for more operations? We love contributions! Feel free to send us your pull requests. Because making Pact testing better is a team sport! 🏆
```